src/algebra/big_operators/finsupp.lean

```diff
@@ -498,6 +498,10 @@ begin
   rw [not_mem_support_iff.mp hi, single_zero],
 end
 
+lemma indicator_const_eq_sum_single [add_comm_monoid M] (s : finset α) (m : M) :
+  indicator s (λ _ _, m) = ∑ x in s, single x m :=
```

Comment:
Are you sure you don't want a special case about `f : α → M` rather? This already lets you get rid of the `attach` in the sum.